### PR TITLE
STACK-790 : Revert tasks for handler_server

### PR DESCRIPTION
**Pull request by NehaKembalkarA10**
**Non-Customer Highlights**
- Added revert functions to handler_server task.

- Added unit tests for the reverts of handler_task.

**Closes**
STACK-790

### CODE REVIEW

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 1]
```diff
@@ -22,87 +22,96 @@
...
+            LOG.exception("Failed to create member: %s", str(e))
+            raise
+
+        try:
+            c.slb.service_group.member.create(
+                pool.id, member.id, member.protocol_port)
+            LOG.debug("Member %s associated to pool %s successfully",
+                      member.id, pool.id)
+        except Exception as e:
+            LOG.exception("Failed to create pool: %s", str(e))
```
Comment (omkartelee01) [inline a10_octavia/controller/worker/tasks/handler_server.py]: LOG.exception(Failed to associate member to pool: %s, str(e))

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 2]
```diff
@@ -22,87 +22,96 @@
...
     def execute(self, member, vthunder, pool):
         """ Execute delete member """
+        c = self.client_factory(vthunder)
         try:
-            c = self.client_factory(vthunder)
-            c.slb.service_group.member.delete(pool.id, member.id, member.protocol_port)
-            LOG.info("Member dissociated from pool successfully.")
+            c.slb.service_group.member.delete(
+                pool.id, member.id, member.protocol_port)
+            LOG.debug("Member dissociated from pool successfully.")
```
Comment (omkartelee01) [inline a10_octavia/controller/worker/tasks/handler_server.py]: LOG.debug("Member %s dissociate from pool %s successfully", member.id, pool.id)

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 3]
```diff
@@ -22,87 +22,96 @@
...
             c.slb.server.delete(member.id)
-            LOG.info("Member deleted successfully.")
+            LOG.debug("Member deleted successfully.")
```
Comment (omkartelee01) [inline a10_octavia/controller/worker/tasks/handler_server.py]: correct log line

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 4]
```diff
@@ -22,87 +22,96 @@
...
-            LOG.error(str(e))
-            LOG.info("Error occurred")
+            LOG.warning("Failed to delete member: %s", str(e))
 
 
 class MemberUpdate(BaseVThunderTask):
     """ Task to update member """
 
     def execute(self, member, vthunder, pool):
         """Execute create member for an amphora."""
```
Comment (omkartelee01) [inline a10_octavia/controller/worker/tasks/handler_server.py]: remove amphora word, correct the function doc string

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 5]
```diff
@@ -22,87 +22,96 @@
...
-            except:
-                server_temp = None
-                LOG.error("Invalid definition of A10 config in Member section.")
-
             c.slb.server.update(member.id, member.ip_address, status=status,
                                 server_templates=server_temp,
                                 axapi_args=server_args)
-            LOG.info("Member updated successfully.")
+            LOG.debug("Member updated successfully: %s", member.id)
+            raise
```
Comment (omkartelee01) [inline a10_octavia/controller/worker/tasks/handler_server.py]: remove this raise

File: a10_octavia/controller/worker/tasks/handler_server.py [hunk 6]
```diff
@@ -22,87 +22,95 @@
...
                                 axapi_args=server_args)
-            LOG.info("Member updated successfully.")
+            LOG.debug("Member updated successfully: %s", member.id)
         except Exception as e:
-            LOG.error(str(e))
-            LOG.info("Error occurred")
+            LOG.exception("Failed to update member: %s", str(e))
+            raise
+
+    def revert(self, member, vthunder, pool, *args, **kwargs):
```
Comment (hthompson6) [inline a10_octavia/controller/worker/tasks/handler_server.py]: We don't need to revert when updating. If it fails then the failure is simply raised.

File: a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+from a10_octavia.common.data_models import VThunder
+from a10_octavia.tests.common import a10constants
+
+VTHUNDER = VThunder()
+POOL = o_data_models.Pool(id=a10constants.MOCK_POOL_ID)
+MEMBER = o_data_models.Member(
+    id=a10constants.MOCK_MEMBER_ID, protocol_port=t_constants.MOCK_PORT_ID)
+
+
+class TestMemberCreate(base.TestCase):
```
Comment (hthompson6) [inline a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py]: This should be `TestHandlerServerTasks(base.TestCase):`

File: a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+    def setUp(self):
+        patcher = patch(
+            'a10_octavia.controller.worker.tasks.common.BaseVThunderTask.client_factory')
+        self.client_factory_mock = patcher.start()
+        self.client_factory_mock.return_value = mock.Mock()
+        super(TestMemberCreate, self).setUp()
+
+    def test_revert_create_server_task(self):
```
Comment (hthompson6) [inline a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py]: `def test_revert_member_create_task(self)`

File: a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+VTHUNDER = VThunder()
+POOL = o_data_models.Pool(id=a10constants.MOCK_POOL_ID)
+MEMBER = o_data_models.Member(
+    id=a10constants.MOCK_MEMBER_ID, protocol_port=t_constants.MOCK_PORT_ID)
+
+
+class TestMemberCreate(base.TestCase):
+
+    def setUp(self):
+        patcher = patch(
```
Comment (hthompson6) [inline a10_octavia/tests/unit/controller/worker/tasks/test_handler_server.py]: Instead we could do

```python
mock_patch = patcher.start()
self.client_mock = mock.Mock()
mock_patch.return_value = self.client_mock
```

Then later on we just have to check ` self.client_mock.slb.server.delete.assert_called_with`

**Pull request merged**